### PR TITLE
[IA-2441] Use google2 billing

### PR DESCRIPTION
**Pull request by jdcanas**
This PR removes the use of the google1 dao currently used for billing.

It currently uses a snap, pending this PR: https://github.com/broadinstitute/workbench-libs/pull/490

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala [hunk 1]
```diff
@@ -424,12 +425,10 @@ class DataprocInterpreter[F[_]: Timer: Parallel: ContextShift](
...
-            projectNumberOptIO <- F.liftIO(
-              IO.fromFuture(IO(googleProjectDAO.getProjectNumber(googleProject.value)))
-            )
-            projectNumber <- F.liftIO(
-              IO.fromEither(projectNumberOptIO.toRight(ClusterIamSetupException(imageProject)))
-            )
+            projectNumberOpt <- googleResourceService.getProjectNumber(googleProject)
+
+            //TODO: This can also be from a 404?
+            projectNumber <- F.fromEither(projectNumberOpt.toRight(ClusterIamSetupException(imageProject)))
```
Comment (jdcanas) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala]: So, I'm not really sure in what circumstances the exception here occurs.

I wanted to see if anyone knew what it was for (@rtitle ?) before I touchedd it, but I added a TODO there with my thoughts. I'm thinking this should be a project not found exception. As in, the library returns `Option[Long]` now, where it is none if the project passed is not found. Any idea if the exception here is still relevant?

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala]: my guess is this was just for dealing with the case when project doesn't exist...does your `googleResourceService.getProjectNumber` return `None` when project doesn't exist currently?

Comment (jdcanas) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala]: yeah, I'm thinking I should change the exception here

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/DataprocInterpreter.scala]: @rtitle just wonder if you have more context around this code...

**Review by Qi77Qi — APPROVED** [review]
lgtm as long as the exception refactor is what was intended

**Pull request merged**